test/basic.js

```diff
@@ -128,6 +130,31 @@ t.test('--harmony', function (t) {
   })
 })
 
+t.test('node exe with different name', function(t) {
+  var fp = path.join(__dirname, 'fixtures', 'exething')
+  fs.createReadStream(process.execPath)
```

Comment:
I tried that at first and it wasn't working properly. I believe `process.execPath` is the resolved path.